src/module.c

```diff
@@ -2049,9 +2049,13 @@ void RM_Yield(RedisModuleCtx *ctx, int flags, const char *busy_reply) {
                 blockingOperationStarts();
             }
 
+            protectClient(ctx->client);
+
             /* Let redis process events */
             processEventsWhileBlocked();
 
+            unprotectClient(ctx->client);
```

Comment:
i think we wanna move the protectClient to be next to blockingOperationStarts and the unprotect to be next to the blockingOperationEnds call, so that we don't protect and unprotect on each RM_Yield call.